src/plasma/plasma_manager.cc

```diff
@@ -235,6 +235,11 @@ struct PlasmaManagerState {
   /** The time (in milliseconds since the Unix epoch) when the most recent
    *  heartbeat was sent. */
   int64_t previous_heartbeat_time;
+  /**
+   * Objects are mapped to the time, in milliseconds, they were received. Objects
+   * are removed from this map after they are added to local_available_objects.
+   */
+  std::unordered_map<ObjectID, int64_t, UniqueIDHasher> received_objects;
```

Comment:
If I understand correctly, this map contains precisely the objects where the transfer has been completed but the store hasn't sent a notification to the manager yet. Is that right?

Reply:
yes.